### ISSUE TITLE
Consider pulling back in some updates here?

### ISSUE THREAD

**Issue by acorncom** [offirgolan/ember-data-copyable#58]
@miguelcobain noticed that you'd been given commit rights here. Wondered if you might be interested in some of the work from https://github.com/eflexsystems/ember-data-copyable ? Looks like @jakesjews did a bunch of work there that might be useful to roll back into the main community add-on ...

**Comment by miguelcobain** [comment]
Hey @acorncom . I modernized this project a bit (to latest 3.x blueprint), including the tests and some dependencies. It was really difficult to make newer ember-cli-mirage + ember-data happy with the way we define models dynamically, but I made it.

From the branch you referred, I included support for `optionsPropertyName`. Is there anything else you wanted from that branch that isn't on the main one?
